src/components/molecules/modals/reuploadModal/ReuploadModal.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import './reupload.scss';
+import FileSelector from 'components/atoms/fileSelector/FileSelector';
+import ConfirmationBox from 'components/molecules/confirmationBox/ConfirmationBox';
+import Modal from 'components/molecules/modal/Modal';
+import useModal from 'hooks/useModal';
+import { useToast } from 'hooks/useToast';
+import { useHistory } from 'react-router-dom';
+import { PNGImageValidator } from 'utils/validators';
+
+interface IReuploadImageModal {
+  title: string;
+  id: string | undefined;
+  shouldOpen: boolean;
+  prefix: string;
+  uploadFunction: (id: string, file: File) => Promise<void>;
+}
+
+const ReuploadImageModal: React.FC<IReuploadImageModal> = ({
+  title,
+  id,
+  shouldOpen,
+  prefix,
+  uploadFunction,
+}) => {
+  const [file, setFile] = useState<File | undefined>();
+  const [error, setError] = useState<string | undefined>(undefined);
+  const { addToast } = useToast();
+  const history = useHistory();
+  const { isOpen, onOpen, onClose } = useModal();
+
+  const reuploadImage = async (id: string) => {
+    if (!file) {
+      setError('Bildefil ikke valgt');
+      return;
+    }
+    setError('');
+    try {
+      await uploadFunction(id, file);
+    } catch (error) {
+      addToast({
+        title: 'Feil ved opplasting av bilde',
+        status: 'error',
+        description: 'prøv på nytt',
+      });
+      return;
+    }
+    redirAndDispalyFeedback();
+  };
+
+  const redirAndDispalyFeedback = () => {
+    addToast({
+      title: 'Suksess',
+      status: 'success',
+      description: 'Bildet er lagt til',
+    });
+    history.push(`${prefix}/${id}`);
+  };
+
+  useEffect(() => {
+    if (shouldOpen) {
+      onOpen();
+    }
+  }, [shouldOpen]);
+
+  useEffect(() => {
+    if (!isOpen && id !== undefined) {
+      // handles modal closed and users has not uploaded event
+      redirAndDispalyFeedback();
+      return;
+    }
+  }, [isOpen]);
+
+  return (
+    <Modal title={title} isOpen={isOpen} onClose={onClose}>
+      {id !== undefined && (
+        <div className="imgReupload">
+          <FileSelector
+            setFile={setFile}
+            text="En feil skjedde ved opplasting av bilde, prøv på nytt her."
+            accept="image/png"
+            fileValidator={PNGImageValidator}
+          />
+          <ConfirmationBox
+            confirmText="Last opp på nytt"
+            declineText="Ikke last opp nytt bilde"
+            onAccept={() => {
+              reuploadImage(id);
+            }}
+            onDecline={() => {
+              onClose();
+              redirAndDispalyFeedback();
```

Comment:
Does it show success that the picture id added on decline?

Reply:
fixed now :+1: